PowerEditor/src/Notepad_plus_Window.cpp

```diff
@@ -385,12 +390,13 @@ void Notepad_plus_Window::init(HINSTANCE hInst, HWND parent, const TCHAR *cmdLin
 
 	if (cmdLineParams->_showLoadingTime)
 	{
-		time_t timestampEnd = time(NULL);
-		double loadTime = difftime(timestampEnd, timestampBegin);
-
-		char dest[256];
-		sprintf(dest, "Loading time : %.0lf seconds", loadTime);
-		::MessageBoxA(NULL, dest, "", MB_OK);
+		std::chrono::steady_clock::duration nppInitTime = (std::chrono::steady_clock::now() - g_nppStartTimePoint) - sessionLoadingTime - cmdlineParamsLoadingTime;
+		std::wstringstream wss;
+		wss << L"app & plugin(s) initialization: " << std::chrono::hh_mm_ss{ std::chrono::duration_cast<std::chrono::milliseconds>(nppInitTime) } << std::endl;
+		wss << L"last session loading: " << std::chrono::hh_mm_ss{ std::chrono::duration_cast<std::chrono::milliseconds>(sessionLoadingTime) } << std::endl;
+		wss << L"cmdline params handling: " << std::chrono::hh_mm_ss{ std::chrono::duration_cast<std::chrono::milliseconds>(cmdlineParamsLoadingTime) } << std::endl << std::endl;
+		wss << L"overall: " << std::chrono::hh_mm_ss{ std::chrono::duration_cast<std::chrono::milliseconds>(nppInitTime + sessionLoadingTime + cmdlineParamsLoadingTime) };
+		::MessageBoxW(NULL, wss.str().c_str(), L"Notepad++ loading time (hh:mm:ss.ms)", MB_OK);

```

Comment:
Very good PR. But if we can get plugins loading time, it'll be even better.
How about the following message:
```
Notepad++ initialization: 0:0:0.007
Plugins loading: 0:0:0.666
Last session loading: 0:0:0.042
Command line params handling: 0:0:0.069
Total loading time: 0:0:0.784
```
Since you have already the number of *"app & plugin(s) initialization:"*, you can get **"Notepad++ initialization:"** easily if you have also "Plugins loading:":
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/Notepad_plus.cpp#L460

What do you think?

Reply:
I like both the message and the addition of separate time measurement for plugins.

I have already considered some others (e.g. to have exact pure file loading time instead of the sessionTime and the cmdlineTime measurements) but I think that the current state of PR is enough (+ the pluginsTime).
We can always add or refine some criteria later.